src/main/java/org/zeromq/ZMQ.java

```diff
@@ -757,9 +757,22 @@ public void close()
          * @param type    the socket type.
          */
         protected Socket(Context context, SocketType type)
+        {
+            this(context, type.type);
```

Comment:
I Think to reference a Deprecated method is not a good idea, I prefer reverse of this, deprecated method call newly created method as in this situation removing deprecated method has no side effect

Reply:
It's internal change, this one is faster, as the int type is already available. If you swap constructors callings, code will have to iterate each time to find out the proper enum.

Reply:
I understand, but creating a socket is not in the critical path.

Reply:
Of course, but for the sake of compatibility with previous releases, this constructor will stay in place for several versions. Doing it that way is a bit faster, so I propose to stay pragmatic and use it for the time being. When the time will come to remove the deprecated constructor, we can recode the other.